shared/validate/validate.go

```diff
@@ -755,6 +757,34 @@ func IsListenAddress(allowDNS bool, allowWildcard bool, requirePort bool) func(v
 	}
 }
 
+// IsListenAddressList returns a validator for a set of comma-separated addresses.
+func IsListenAddressList(allowDNS bool, allowWildcard bool, requirePort bool) func(value string) error {
+	return func(value string) error {
+		for _, v := range strings.Split(value, ",") {
+			v = strings.TrimSpace(v)
+			f := IsListenAddress(allowDNS, allowWildcard, requirePort)
+			err := f(v)
+			if err != nil {
+				return err
+			}
+		}
+
+		return nil
+	}
+}
+
+// IsX509Certificate checks if the value is a valid x509 PEM Certificate.
+func IsX509Certificate(value string) error {
+	certBlock, _ := pem.Decode([]byte(value))
+	if certBlock == nil {
+		return fmt.Errorf("Received Invalid certificate")
```

Comment:
Please can we just use "Invalid..." Error here rather than prefixing with "Received" for consistency with other validators?

As we don't know in what context the validator will be called in and that may not always be appropriate.